File: app/src/main/java/com/mxt/anitrend/base/custom/animation/ScaleAnimation.java
package com.mxt.anitrend.base.custom.animation;

import android.animation.Animator;
import android.animation.ObjectAnimator;
import android.view.View;
import android.view.animation.Interpolator;
import android.view.animation.LinearInterpolator;

import com.mxt.anitrend.base.interfaces.base.BaseAnimation;

/**
 * Created by max on 2018/02/24.
 */

public class ScaleAnimation implements BaseAnimation {

    private final float FROM, TO;
    private Interpolator interpolator = new LinearInterpolator();

    public ScaleAnimation() {
        this.FROM = .85f;
        this.TO = 1f;
    }

    public ScaleAnimation(float FROM, float TO) {
        this.FROM = FROM;
        this.TO = TO;
    }

    @Override
    public Animator[] getAnimators(View view) {
        ObjectAnimator scaleX = ObjectAnimator.ofFloat(view, "scaleX", FROM, TO);
        ObjectAnimator scaleY = ObjectAnimator.ofFloat(view, "scaleY", FROM, TO);
        return new Animator[]{scaleX, scaleY};
    }

    @Override
    public Interpolator getInterpolator() {
        return interpolator;
    }

    @Override
    public int getAnimationDuration() {
        return DEFAULT_ANIMATION_DURATION;
    }
}

File: app/src/main/java/com/mxt/anitrend/base/custom/animation/ScaleAnimation.kt
package com.mxt.anitrend.base.custom.animation

import android.animation.Animator
import android.animation.ObjectAnimator
import android.view.View
import android.view.animation.Interpolator
import android.view.animation.LinearInterpolator
import com.mxt.anitrend.base.interfaces.base.BaseAnimation

/**
 * Created by max on 2018/02/24.
 */
class ScaleAnimation
@JvmOverloads
constructor(
    private val from: Float = 0.85f,
    private val to: Float = 1f,
) : BaseAnimation {
    private val interpolator: Interpolator = LinearInterpolator()

    override fun getAnimators(view: View): Array<Animator> {
        val scaleX = ObjectAnimator.ofFloat(view, "scaleX", from, to)
        val scaleY = ObjectAnimator.ofFloat(view, "scaleY", from, to)
        return arrayOf(scaleX, scaleY)
    }

    override fun getInterpolator(): Interpolator = interpolator

    override fun getAnimationDuration(): Int = BaseAnimation.DEFAULT_ANIMATION_DURATION
}
